drivers/axi_core/spi_engine/spi_engine.h

```diff
@@ -177,6 +177,15 @@ struct spi_engine_offload_message {
  */
 extern const struct spi_platform_ops xil_platform_ops;
 
+/* Write SPI Engine's axi registers */
+int32_t spi_engine_write(struct spi_engine_desc *desc,
```

Comment:
Why isn't used spi_platform_ops?

Reply:
Because spi_platform_ops supports only `spi_write_and_read` functions.

The `spi_engine_write` function is used explicitly in the `ad5766_core` has different implementation than the `write_and_read` function.

Reply:
Maybe something about this can be added in the commit message

Reply:
There is already something added in the commit message. But I will try to make it more explicit.

Reply:
Maybe add `spi_engine_read` and `spi_engine_write` to `struct spi_platform_ops`?

Reply:
or something like:
```
struct spi_message {
     uint8_t *tx_buff,
     uint8_t *rx_buff,
     uint32_t tx_len,
     uint32_t rx_len
}

int32_t spi_transfer(struct spi_desc *desc, struct spi_message *msg);
```

Reply:
See please the note in the PR description.
Is it worth complicating the spi implementation? Since the ad5766_core files which are using spi_engine_write and spi_engine_read will be removed and these suggested improvements won't be used anymore.

I find the current approach being the easiest to handle, taking into account the current circumstances.